### DISCUSSION TITLE
how to get the current issue buffer octo options? i need the orga/repo string

### DISCUSSION THREAD

**Discussion by divramod** [pwntester/octo.nvim#365] [Q&A]
hey hey,

i want to write a custom function, i can use, when i am in a octo issue buffer.
it would be nice to know how to access the metadata of the issue.
i suspect they are saved somewhere in the buffer options?

i can get the current buffers filetype like that
```
    local current_win = vim.api.nvim_get_current_win()
    local current_buf = vim.api.nvim_win_get_buf(current_win)
    local filetype = vim.api.nvim_buf_get_option(current_buf, 'filetype')
    print(filetype)
```

is something like this possible?:

```
    local current_win = vim.api.nvim_get_current_win()
    local current_buf = vim.api.nvim_win_get_buf(current_win)
    local octo = vim.api.nvim_buf_get_option(current_buf, 'octo')
    print(octo.repo)
```

thx btw for the awesome plugin!

i use it every day.

**Comment by pwntester** [comment]
Hi, thanks for the kind words!

You can get the nwo (name with owner) from `require("octo.utils").get_remote_name()`. You can check how Im using it on my [lualine config](https://github.com/pwntester/octo.nvim/pull/367).

**Reply by divramod** [reply]
hey hey,

thx for the fast reply.

sadly ```require("octo.utils").get_remote_name()``` is not enough for me.
i am organizing my todolist in a git repo. if i open a file in that repo and then open a octo buffer, ```require("octo.utils").get_remote_name()``` returns the reponame of my markdown files and not the repo name of the issue.

my folder structure
```
my-markdown-repo/
  .git/
  README.md
```

after that i do
- ```:Octo https://github.com/orga/another_repo/issues/48```
- after that i call the function below
- i just started using lua, so i am a beginner (but i like lua)
- its working until print('4 is buffer'), when i am in a octo buffer. i think, i am really close.

```
_G.DmO = function()
  local winNr = vim.api.nvim_get_current_win()
  local bufNr = vim.api.nvim_win_get_buf(winNr)
  print('1 bufNr: ' .. bufNr)
  if vim.api.nvim_buf_get_option(bufNr, "filetype") == "octo" then
    print('2 is octo')
    if type(_G.octo_buffers) == "table" then
      print('3 is type octo_buffers')
      local buffer = _G.octo_buffers[bufNr]
      if buffer then
        print('4 is buffer')
        buffer:update_metadata()
        for key, val in ipairs(buffer.titleMetadata) do
          print(key .. ": " .. val)
        end
        for key, val in ipairs(buffer.bodyMetadata) do
          print(key .. ": " .. val)
        end
      end
    end
  end
end
```

1. it would be nice to know for me how to access the title, body, comment, etc metadata of the octo buffer (are there other metadata objects?).
2. is there a list, on which metadata keys are in existance in these objects?
3. i suspect the octo-buffer-issues repo name is somewhere in there. can you tell how to access it?

**Reply by pwntester** [reply]
Sure, take a look at the `lua/octo/model/octo-buffer.lua` class. When you get one buffer from the global `octo_buffers` table, you will get an instance of that class. In there, you will see a `repo` field. So you can get the repo name for that buffer from:

```lua
local nwo = buffer.repo
```

As for the fields for the different metadata classes, you can check them in the `lua/octo/model/xxx-metadata.lua` classes

Cheers,
A
